stdlib/messaging/kafka/src/main/java/org/ballerinalang/messaging/kafka/impl/KafkaListenerImpl.java

```diff
@@ -59,8 +70,9 @@ public KafkaListenerImpl(Strand strand, ObjectValue listener, ObjectValue servic
     @Override
     public void onRecordsReceived(ConsumerRecords records, KafkaConsumer kafkaConsumer, String groupId) {
         listener.addNativeData(NATIVE_CONSUMER, kafkaConsumer);
-        Executor.submit(this.scheduler, service, KAFKA_RESOURCE_ON_MESSAGE, callback,
-                null, getResourceParameters(service, this.listener, records, groupId));
+        String url = ((Properties) listener.getNativeData(NATIVE_CONSUMER_CONFIG)).getProperty(BOOTSTRAP_SERVERS);
```

Comment:
Is NATIVE_CONSUMER_CONFIG always present? Cause otherwise, `getNativeData` can return Null. Therefore possible NPE.